Ginger/GingerCore/GingerOCR/GingerOcrOperations.cs

```diff
@@ -0,0 +1,522 @@
+                    {
+                        txtOutput = string.Concat(txtOutput, ReadTextFromByteArray(pngByte));
+                    }
+                }
+                else
+                {
+                    Reporter.ToUser(eUserMsgKey.StaticErrorMessage, "Invalid Page Number Provided");
+                    Reporter.ToLog(eLogLevel.ERROR, "Invalid Page Number Provided");
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, ex.Message, ex);
+            }
+            return txtOutput;
+        }
+
+        private static Page GetPageObjectFromByteArray(byte[] byteArray)
+        {
+            try
+            {
+                using (Pix img = Pix.LoadFromMemory(byteArray))
+                {
+                    Page page = Instance.Process(img);
+                    return page;
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, ex.Message, ex);
+            }
+            return null;
+        }
+
+        private static Page GetPageObjectFromFilePath(string filePath)
+        {
+            try
+            {
+                using (Pix img = Pix.LoadFromFile(filePath))
+                {
+                    return Instance.Process(img);
+                }
+            }
+            catch (Exception ex)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, ex.Message, ex);
+            }
+            return null;
+        }
+
+        public static string ReadTextAfterLabelPdf(string pdfFilePath, string label, string pageNum = "", string password = null)
+        {
+            string resultTxt = string.Empty;
+            if (!string.IsNullOrEmpty(pageNum))
+            {
+                List<byte[]> lstPngByte = GetPngByteArrayFromPdf(pdfFilePath, pageNum, password);
+
+                foreach (byte[] pngByte in lstPngByte)
+                {
+                    resultTxt = GetResultTextFromByteArray(label, resultTxt, pngByte);
+                    if (!string.IsNullOrEmpty(resultTxt))
+                    {
+                        break;
+                    }
+                }
+            }
+            else
+            {
+                List<byte[]> lstByteArray = GetListOfPngByteArrayFromPdf(pdfFilePath, password);
+                foreach (byte[] byteArray in lstByteArray)
+                {
+                    resultTxt = GetResultTextFromByteArray(label, resultTxt, byteArray);
+                    if (!string.IsNullOrEmpty(resultTxt))
+                    {
+                        break;
+                    }
+                }
+            }
+            return resultTxt;
+        }
+
+        private static string GetResultTextFromByteArray(string label, string resultTxt, byte[] byteArray)
+        {
+            Page pageObj = GetPageObjectFromByteArray(byteArray);
+            using (pageObj)
```

Comment:
can we create pageObj inside using, if not, lets assign null to in finally block so garbage collector will free its memory